test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Validators/FilterQueryValidatorTest.cs

```diff
@@ -1363,7 +1363,7 @@ private static TheoryDataSet<string> GetLongInputsTestData(int maxCount)
                     "" + String.Join(" add ", Enumerable.Range(1, maxCount/2)) + " eq 5050",
                     "" + String.Join("/", Enumerable.Range(1, maxCount/2).Select(_ => "Category/Product")) + "/ProductID eq 1",
                     "" + String.Join("/", Enumerable.Range(1, maxCount/2).Select(_ => "Category/Product")) + "/UnsignedReorderLevel eq 1",
-                    "" + Enumerable.Range(1,maxCount).Aggregate("'abc'", (prev,i) => String.Format("trim({0})", prev)) + " eq '123'",
+                    //"" + Enumerable.Range(1,maxCount).Aggregate("'abc'", (prev,i) => String.Format("trim({0})", prev)) + " eq '123'", // Recursion Depth exceeded in Microsoft.OData.Core
```

Comment:
>// Recursion Depth exceeded in Microsoft.OData.Core [](start = 136, length = 51)

It's a ODL regression?

Reply:
No it's not
The `ParseInHas()` method in this PR https://github.com/OData/odata.net/pull/2017 added an extra recursion. Since this test is performing many recursion calls, we are exceeding the default recursion depth limit of `800`.

ps: Before the Fix, this test had a maximum recursionDepth of `700`.